tests/rptest/tests/data_migrations_api_test.py

```diff
@@ -27,6 +30,39 @@
 import requests
 
 
+class TransferLeadersBackgroundThread:
+    def __init__(self, redpanda: RedpandaServiceBase, topic: str):
+        self.redpanda = redpanda
+        self.logger = redpanda.logger
+        self.stop_ev = threading.Event()
+        self.topic = topic
+        self.admin = Admin(self.redpanda)
+        self.thread = threading.Thread(target=lambda: self._loop())
+        self.thread.daemon = True
+
+    def start(self):
+        self.thread.start()
+
+    def stop(self):
+        self.stop_ev.set()
+
+    def _loop(self):
+        while not self.stop_ev.is_set():
+            partitions = self.admin.get_partitions(namespace="kafka",
+                                                   topic=self.topic)
+            partition = random.choice(partitions)
+            p_id = partition['partition_id']
+            self.logger.info(f"Transferring leadership of {self.topic}/{p_id}")
+            try:
+                self.admin.partition_transfer_leadership(namespace="kafka",
+                                                         topic=self.topic,
+                                                         partition=p_id)
+            except Exception as e:
+                self.logger.info(
+                    f"error transferring leadership of {self.topic}/{p_id} - {e}"
+                )
```

Comment:
I guess it's for a situation where a topic is already gone or not yet created?

Reply:
yes, exactly or timeouts

Reply:
yes, exactly or timeouts